### PR TITLE
dictionary needs to be able to save when there are no entries in the shared section

### PR DESCRIPTION
**Pull request by GregHanson**
I am working on incorporating the w3c [rdf-tests TTL samples files](https://github.com/w3c/rdf-tests/tree/main/sparql/sparql10) to ensure the covnerter handles known rdf syntaxes similar to what was done in the `oxhdt-sys` 
I incorporated into my initial upstream oxigraph PR https://github.com/oxigraph/oxigraph/pull/1211/commits/40a3467e8c2163c261084bef48c5e14b1214cf46 

While running those files through the converter, found a bug when one of the 4 section dictionaries did not have any entries (shared specifically) using https://github.com/w3c/rdf-tests/blob/main/sparql/sparql10/basic/data-3.ttl as an example. I can't remember if the assert was needed originally and became unnecessary as I iterated the `pack_bits` function, or whether I was just being overly cautious - but the assert is apparently no longer needed and actually breaks for the edge case described above

### CODE REVIEW

**Comment by GregHanson** [comment]
I'm not sure if you want a test for this edge case in this PR, or if you are OK with me including it as part of https://github.com/KonradHoeffner/hdt/pull/56

**Comment by GregHanson** [comment]
/hold on this one for now - found another potential while running through the rdf-tests files:
[rdf-tests/sparql/sparql11/property-path/empty.ttl](https://github.com/w3c/rdf-tests/blob/7d298cbc1de7995ecb5accabda3e1f5c9505291d/sparql/sparql11/property-path/empty.ttl)

error occurs after attempting to load the converted hdt file:
```bash
thread 'rdf2hdt::builder::tests::test_sparql11' panicked at src/triples.rs:346:55:
called `Option::unwrap()` on a `None` value
```

**Comment by GregHanson** [comment]
> [rdf-tests/sparql/sparql11/property-path/empty.ttl](https://github.com/w3c/rdf-tests/blob/7d298cbc1de7995ecb5accabda3e1f5c9505291d/sparql/sparql11/property-path/empty.ttl)

@KonradHoeffner I'm not sure of the expected behavior - but is an HDT file valid if the original RDF data has 0 triples? I tried the hdt-cpp version of the rdf2hdt tool and it does produce a HDT file without error, and `hdtSearch` is able to load it. The library as it stands (regardless of my rdf2hdt changes) cannot open an HDT file with 0 triples due to this `.max().unwrap()` call.

1. should the hdt crate be able to open HDT files with 0 triples?
2. should the rdf2hdt crate create HDT files with 0 triples or should it error out when that scenario is detected?

**Comment by KonradHoeffner** [comment]
>     1. should the hdt crate be able to open HDT files with 0 triples?
Theoretically I could imagine HDT files with 0 triples as some kind of theoretical concept for some specification to make sense or some proof by induction or something similar. 
But practically I have never needed this and cannot imagine a scenario where it would be useful to load an HDT file with 0 triples.
In fact, as a user I would rather get an error so because something certainly has gone wrong.
The only scenario I can think of is for a mutable RDF graph to be created empty and then being filled step by step but because HDT graphs are immutable this never occurs. 
That said, panicking is bad and I am glad you brought this up, so we can create a more descriptive error.

>     2. should the rdf2hdt crate create HDT files with 0 triples or should it error out when that scenario is detected?

I think it should error out for the same reason that HDT is read-only.

File: src/containers/sequence.rs [hunk 1]
```diff
@@ -214,8 +214,6 @@ impl Sequence {
     }
 
     fn pack_bits(&self) -> Vec<u8> {
-        assert!(self.bits_per_entry > 0 && self.bits_per_entry <= std::mem::size_of::<usize>() * 8);
```
Comment (KonradHoeffner) [inline src/containers/sequence.rs:217]: Removing the line completely would however skip the upper bounds check as well.

Comment (KonradHoeffner) [reply src/containers/sequence.rs:217]: Oh, this is only used in your save function.

**Comment by KonradHoeffner** [comment]
To make sure that this isn't an issue for the reading process as well I also tested loading a file with no shared section but that worked without issues.

**Comment by GregHanson** [comment]
I added a bunch of the rdf-tests in my rdf2hdt PR here https://github.com/KonradHoeffner/hdt/commit/40b0041521fb0af72925ad31dc43ff65f2cf59ab

Just verifying that each of those w3c examples convert and then successfully load

**Comment by GregHanson** [comment]
> The only scenario I can think of is for a mutable RDF graph to be created empty and then being filled step by step but because HDT graphs are immutable this never occurs.

This makes sense, since if I am reading code right, the C++ version of the tooling allows for optional mutable HDT.

Since mutable HDT is outside the scope of the existing HDT crate, and our use cases as well, I completely agree with erroring out in the 0 triples scenario

**Comment by KonradHoeffner** [comment]
> This makes sense, since if I am reading code right, the C++ version of the tooling allows for optional mutable HDT.
> 
That is surprising to me as the spec website describes it as a read only format.
